core/src/avm2/object.rs

```diff
@@ -212,10 +212,9 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
             }
         }
 
-        let has_no_getter =
-            self.has_own_virtual_setter(&name) && !self.has_own_virtual_getter(&name);
+        let is_set_only = self.base().has_own_virtual_set_only_property(&name);
 
-        if self.has_own_property(&name)? && !has_no_getter {
+        if self.has_own_property(&name)? && !is_set_only {
```

Comment:
In testing on my branch I was able to elide this check entirely (`avm2-multinameperf`) without failing any tests. It used to be needed back when trait resolution went through prototypes; but no longer.

If this is intended to throw an error when getting a set-only property, I think this may be a little too late. Specifically, it's relying on `proto` doing a duplicate `resolve_multiname` and getting `None`. If there is no `proto`, then no error is thrown. This may not be that big of a deal, though.

In the future, if we get rid of `Property::Virtual`, and do a lookup for getter traits directly, then we don't need to check for set-only properties at all. That's something I haven't implemented yet, but have been thinking about.

Reply:
...it just occured to me that we really should be calling `get_property_undef` instead of returning `Value::Undefined` at the bottom of this function. That would also fix this very-theoretical problem and fix an edge case in `Proxy`. (at least, I'm pretty sure set-only properties are supposed to call `flash_proxy::getproperty`?)

Reply:
> at least, I'm pretty sure set-only properties are supposed to call Proxy.flash_proxy::getproperty?

I don't think so; the logic in FP is closer to:
```
get_property(obj, multiname):
    // (checks for int/dictionary lookups occur earlier, directly in opcode implementation; this is a generic getproperty tool)
    var property = find_in_traits(obj, multiname);
    if (property is field or method or get-property or get-set-property)
        normal path;
    else if (property is set-property)
        throw error;
    else if (property not found)
        fallback to virtual obj->get_property(multiname) (this activates Proxy machinery)
```

And indeed:
```js
class C extends Proxy {
    public function set x(v) { trace("setter") }
    flash_proxy override function getProperty(name:*):* { trace("via proxy") }
}
var c = new C();
trace(c.x);
```
throws:
```
ReferenceError: Error #1077: Illegal read of write-only property x on FilePrivateNS:Test.C.
	at global$init()
```
Same with setting read-only props.

Reply:
Gotcha. I'll refrain from removing this check then.